File: sms.py
import os
from urllib.parse import urlencode
import requests


def sendSMS(phone: str, message: str, linkID: str):
    url = os.environ['SMS_API_URL']
    apiKey = os.environ['SMS_API_KEY']
    userName = os.environ['SMS_API_USERNAME']

    data = urlencode({
        "username": userName,
        "to": phone,
        "message": message.replace('\n', ' '),
        "from": os.environ.get('SHORT_CODE', '2202'),
        "linkId": linkID,
    })

    headers = {
        "Accept": "application/json",
        "Content-Type": "application/x-www-form-urlencoded",
        "apikey": apiKey,
    }

    res = requests.post(url, data=data, headers=headers)
    if res.status_code != 201:
        print(f"Error sending SMS: {res.text}")
        return False

    print("SMS sent successfully", res.text)
    return True

File: ussd/src/service/sms.ts
import axios from "axios";
import FormData from "form-data";

const sendSMS = async (
  phone: string,
  message: string,
  linkID = ""
): Promise<void> => {
  try {
    const apiKey = process.env.SMS_API_KEY;
    const userName = process.env.SMS_API_USERNAME;
    const from = process.env.SHORT_CODE;

    const data = new FormData();
    data.append("username", userName);
    data.append("to", phone);
    data.append("message", message);
    data.append("from", from);

    const options = {
      method: "POST",
      headers: {
        accept: "application/json",
        "content-type": "application/x-www-form-urlencoded",
        apikey: apiKey,
      },
      data,
      url: process.env.SMS_API_URL,
    };

    await axios(options);
  } catch (e) {
    console.log("error sending message", e.message);
  }
};

export default sendSMS;
